algorithms/linfa-pls/src/pls_generic.rs

```diff
@@ -313,12 +313,17 @@ impl<F: Float> PlsValidParams<F> {
         let eps = F::epsilon();
 
         let mut y_score = Array1::ones(y.ncols());
+        let mut found = false;
```

Comment:
Instead of introducing `found`, you can make `y_score` into an `Option` that defaults to None and do `let y_score = y_score.ok_or(PlsError::PowerMethodConstantResidualError())?;`

Reply:
Sure! Defenitely the Rust way, thanks.